### ISSUE TITLE
Regularizing Generative Adversarial Networks under Limited Data

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#153]
## TL; DR

- Add LeCam divergence regularization on discriminator loss against exponential moving average logit.

## Motivation

- LeCam divergence is robust against extreme values, which are likely to be observed in low-shot GAN training.
- Exponential moving average parameter stabilizes the overall training.

## Prior Works

- Augmentations (ADA, DiffAug)
- Regularization (GP, distanceGAN variant)

## Method

![image](https://user-images.githubusercontent.com/46308407/152492050-82b37e0e-c0f1-4205-8aae-d04f3c6fee76.png)
![image](https://user-images.githubusercontent.com/46308407/152492094-331c6cf8-2dfc-4c0f-a547-d6bcead65982.png)
![image](https://user-images.githubusercontent.com/46308407/152492111-8d9605f6-b118-4c1e-8a39-0b73f91df28e.png)
![image](https://user-images.githubusercontent.com/46308407/152492122-3addc9a6-8d91-46f9-add1-5565bd17f904.png)
![image](https://user-images.githubusercontent.com/46308407/152492142-e5ff2cb8-6b59-4423-9a6e-6f5680d32a9d.png)

- Some theoretical works on f-divergence and advantage of using LeCam divergence.

![image](https://user-images.githubusercontent.com/46308407/152492259-7fe0c95d-c910-40e5-bae6-83f12bd26b7e.png)

- **LeCam divergence** (triangular discrimination)
![image](https://user-images.githubusercontent.com/46308407/152492387-170da2e9-af77-447c-a92c-9d5bcc98bb53.png)

## Experiments

![image](https://user-images.githubusercontent.com/46308407/152492231-fbd13205-6e30-455f-a301-3f7adebda2e3.png)
![image](https://user-images.githubusercontent.com/46308407/152492416-0bcbf756-d2af-4b69-8ce5-bb19bdf97fc4.png)
![image](https://user-images.githubusercontent.com/46308407/152492426-8940e1fb-4f6c-410e-93f9-03aff63a9d9b.png)
![image](https://user-images.githubusercontent.com/46308407/152492520-b307d8d5-9af3-425b-b658-4ec781789c92.png)
![image](https://user-images.githubusercontent.com/46308407/152492566-26f75241-0bd0-4a66-8f7d-6ccd2ed8e1b3.png)
![image](https://user-images.githubusercontent.com/46308407/152492583-e750b1c7-a083-47da-89a4-1264a10adcb8.png)
![image](https://user-images.githubusercontent.com/46308407/152492637-b9358cd9-e7be-4578-8f71-4f0d0f870193.png)
![image](https://user-images.githubusercontent.com/46308407/152492644-0888089c-a42c-435d-be1f-d6069932a827.png)
![image](https://user-images.githubusercontent.com/46308407/152492711-103c5f16-86cf-45b1-bc32-cf01be70fe81.png)
![image](https://user-images.githubusercontent.com/46308407/152492730-8d09dd92-2c51-4d3e-b3f6-e4220395ff19.png)
![image](https://user-images.githubusercontent.com/46308407/152492745-f1f488ed-6dc5-4c83-b39f-4ba56ce83bed.png)
![image](https://user-images.githubusercontent.com/46308407/152492813-0a6d95c4-ae51-45b5-b5a5-1c675a8a8292.png)

## Comments

- Impressive volume of experiments. Hard work.
- Proposed LeCam regularization, in my opinion, has the effect of fighting discriminator overfitting.
- With limited data, the discrimination task becomes relatively easy, leading to clear divergence between discriminator logits from real and generated images.
- As the proposed method binds discriminator logit to the anchor crossway (i.e., real-anchor(fake), fake-anchor(real)), it is possible that it has the effect of mitigating sharp divergence and thus providing more stable learning signals to the generator.
